### DISCUSSION TITLE
How do you cancel a flow?

### DISCUSSION THREAD

**Discussion by erondpowell** [directus/directus#15655] [Archive]
I have a flow configured with a Filter (Blocking) Even Hook as the trigger.
I want to cancel a flow based on the value of the data payload collected from the event.

The explanation below was initially given to me for "Filter" events.. but now that I'm expanding the documentation and going over Flows granularly, I can't identify of how to configure a Flow to cancel the database transaction... How does that work?

https://docs.directus.io/configuration/flows/triggers.html#:~:text=or%20prevent%20completion%20outright

**Comment by lukas-schaetzle** [comment]
First you have to use the filter option in the trigger block. Then you can use the script operation and simply throw an Error. That would display this error to the user. If you want to cancel the transaction silently, I think you can change the response data in the trigger block and just leave the corresponding object empty.

**Reply by erondpowell** [reply]
Thanks @Tummerhore 

I haven't really used the script block yet, so I will try that asap.

However, currently whenever I return an empty object in the response body (or try anything similar) it assigns null to the item's field values. So I don't think you can silently cancel an event that way. the event still happens but with all null values. 

@licitdev you got any other tips about this? 😁

**Reply by licitdev** [reply]
@erondpowell The `Condition` operation can be used to cancel the database transaction. Whenever a flow ends with a `reject` path, the triggering event is cancelled.

**Comment by erondpowell** [comment]
Okay so looks like there are two strategies to cancel a flow:

Use a condition operation. Whenever a flow ends with a reject path, the triggering event is cancelled.
Throw an error within a Run Script operation.

**Reply by jakedowns** [reply]
in my case, i'm using a blocking filter flow to implement Upsert logic.
i'm checking to see if the incoming payload matches any records created "today" and if it does, i update the record (then reject)
otherwise, i do nothing and let the create happen

i feel like it would be nice if I could reject/block the original create in a graceful manner, rather than throwing an error
like

what if i used a transform payload operation and changed it to return:
```
{
 "block": true // flag to block original create
 // mock create success by returning id of the record that was altered
 "data": {
   id: "read_data[0].id",
   ...{$trigger.payload}
 }
} 
```

**Comment by ricardovigatti** [comment]
Same use case here, I would like to stop the flow execution without triggering an error. Actually, I would like to return with a success response.

**Reply by FlickDaWist** [reply]
you could try my extension here `npm i @flickdawist/custom_error_handler`

**Comment by FlickDaWist** [comment]
I wrote custom error thrower for directus flow operation check [here](https://github.com/FlickDaWist/directus_custom_error) or install via npm `npm i @flickdawist/custom_error_handler`. This will allow you to cancel the flow with your customized messages and error code
